umake/frameworks/games.py

```diff
@@ -385,3 +387,70 @@ def post_install(self):
                         exec=self.exec_link_name,
                         comment=self.description,
                         categories="Development;IDE;"))
+
+
+class Godot(umake.frameworks.baseinstaller.BaseInstaller):
+
+    def __init__(self, **kwargs):
+        super().__init__(name="Godot",
+                         description=_("Multi-platform 2D and 3D opensource game engine"),
+                         only_on_archs=['i386', 'amd64'],
+                         download_page="https://godotengine.org/download",
+                         desktop_filename="godot.desktop",
+                         required_files_path=["godot"],
+                         dir_to_decompress_in_tarball="",
+                         icon_filename='icon.png',
+                         **kwargs)
+        self.icon_url = os.path.join("https://raw.githubusercontent.com/godotengine/godot/",
+                                     "master",
+                                     self.icon_filename)
+        self.exec_name = ''
+
+    arch_trans = {
+        "amd64": "64",
+        "i386": "32"
+    }
+
+    def parse_download_link(self, line, in_download):
+        """Parse Godot download links"""
+        url = None
+        if '.zip' in line:
+            p = re.search(r'href="([^<]*{}.zip)"'.format(self.arch_trans[get_current_arch()]), line)
+            with suppress(AttributeError):
+                url = p.group(1)
+        return ((url, None), in_download)
+
+    def decompress_and_install(self, fds):
+        """Get Godot executable name from url"""
+        super().decompress_and_install(fds)
+        exec_url = self.download_requests[0].url
+        self.exec_name = os.path.splitext(os.path.basename(exec_url))[0]
+
+    def save_icon(self, download_result):
```

Comment:
This as well..you only need to set the icon in the post_install part

Reply:
The zip file from godot don't has icon file. I need to download from other source.

Reply:
This is good...